app/backend/src/couchers/models.py

```diff
@@ -194,6 +185,52 @@ class User(Base):
+    #                                       | phone = None      |  someone else
+    # ,-----------------,                   | token = None      |  verifies            ,-----------------------,
+    # |  Code Expired   |                   | sent = 1970 or zz |  phone xx            |  Verification Expired |
+    # | phone = xx      |  time passes      | verified = None   | <------,             | phone = xx            |
+    # | token = yy      | <------------,    | attempts = 0      |        |             | token = None          |
+    # | sent = zz (exp.)|              |    '-------------------'        |             | sent = zz             |
+    # | verified = None |              |       V    ^                    +-----------< | verified = ww (exp.)  |
+    # | attempts = 0..2 | >--,         |       |    | ChangePhone("")    |             | attempts = 0          |
+    # '-----------------'    +-------- | ------+----+--------------------+             '-----------------------'
+    #                        |         |       |    | ChangePhone(xx)    |                       ^ time passes
+    #                        |         |       ^    V                    |                       |
+    # ,-----------------,    |         |    ,-------------------,        |             ,-----------------------,
+    # |    Too Many     | >--'         '--< |    Code sent      | >------+             |         Verified      |
+    # | phone = xx      |                   | phone = xx        |        |             | phone = xx            |
+    # | token = yy      | VerifyPhone(wrong)| token = yy        |        '-----------< | token = None          |
+    # | sent = zz       | <------+--------< | sent = zz         |                      | sent = zz             |
+    # | verified = None |        |          | verified = None   | VerifyPhone(correct) | verified = ww         |
+    # | attempts = 3    |        '--------> | attempts = 0..2   | >------------------> | attempts = 0          |
+    # '-----------------'                   '-------------------'                      '-----------------------'
+
+    # randomly generated Luhn 6-digit string
+    phone_verification_token = Column(String(6), nullable=True, server_default=text("NULL"))
+
+    phone_verification_sent = Column(DateTime(timezone=True), nullable=False, server_default=text("to_timestamp(0)"))
+    phone_verification_verified = Column(DateTime(timezone=True), nullable=True, server_default=text("NULL"))
+    phone_verification_attempts = Column(Integer, nullable=False, server_default=text("0"))
+
+    __table_args__ = (
+        # Whenever a phone number is set, it must either be pending verification or already verified.
+        # Exactly one of the following must always be true: not phone, token, verified.
+        CheckConstraint(
+            "(phone IS NULL)::int + (phone_verification_verified IS NOT NULL)::int + (phone_verification_token IS NOT NULL)::int = 1",
```

Comment:
I think it's cleaner to write this as you wrote in the comment:

`(phone IS NULL) OR (phone_verification_verified IS NOT NULL) OR (phone_verification_token IS NOT NULL)`

Reply:
It is not as powerful, doesn't catch (!phone && verified)

Reply:
Ah yeah, you're right. Sorry!